compiler-rt/lib/asan/asan_errors.cpp

```diff
@@ -600,6 +602,43 @@ static void PrintShadowMemoryForAddress(uptr addr) {
   Printf("%s", str.data());
 }
 
+static void CheckPoisonRecords(uptr addr) {
+  if (!AddrIsInMem(addr))
+    return;
+  uptr shadow_addr = MemToShadow(addr);
+  unsigned char poison_magic = *(reinterpret_cast<u8 *>(shadow_addr));
+
+  if (poison_magic != kAsanUserPoisonedMemoryMagic)
```

Comment:
if it's the megic, but track_poison is 0, suggest to rerun with the flag?

Reply:
Added

Reply:
probably we need to find the first poisoned byte in the (addr, access_size) and 
lookup by that.

For partial granules, the next byte must be the magic.

Reply:
Actually it must be already bad address, see CHECK_SMALL_REGION :(
then access_size is not very relevant
so just check if it's partial and check the next shadow if needed.

Reply:
Nice catch! Fixed. I also added a partial granule test case (compiler-rt/test/asan/TestCases/use-after-poison-history-size-partial-granule.cpp): https://github.com/llvm/llvm-project/pull/133175/commits/f4c7b3f8cb5346529647f6aa4dd0410ef05130a0